odoo/addons/base/models/ir_mail_server.py

```diff
@@ -524,3 +669,21 @@ def _onchange_encryption(self):
         else:
             self.smtp_port = 25
         return result
+
+    @api.onchange('smtp_authentication')
+    def _onchange_smtp_authentication(self):
+        if self.smtp_authentication != 'certificate':
+            self.smtp_ssl_certificate = False
+            self.smtp_ssl_private_key = False
+
+        elif self.smtp_authentication != 'login':
+            self.smtp_user = False
+            self.smtp_pass = False
```

Comment:
That seems rather awful from a usability standpoint, if the user messes it completely clears the field and they have to re-setup the entire thing. Maybe these should only be cleared on creation / saving based on the auth?